src/libraries/System.Net.Quic/src/System/Net/Quic/Implementations/MsQuic/MsQuicConnection.cs

```diff
@@ -66,23 +66,34 @@ internal sealed class State
 
             public void RemoveStream(MsQuicStream stream)
             {
+                bool closing;
                 lock (this)
                 {
                     StreamCount--;
+                    closing = _closingGCHandle != IntPtr.Zero && StreamCount == 0;
                 }
 
-                if (Closing && StreamCount == 0)
+                if (closing)
                 {
                     Handle?.Dispose();
-                    //if (_stateHandle.IsAllocated) _stateHandle.Free();
+                    try
+                    {
+                        GCHandle gcHandle = GCHandle.FromIntPtr(_closingGCHandle);
+                        Debug.Assert(gcHandle.IsAllocated);
+                        if (gcHandle.IsAllocated) gcHandle.Free();
```

Comment:
```suggestion
                        gcHandle.Free();
```

Reply:
is this generally safe? We have other places with identical check.

Reply:
IsAllocated just checks whether the GCHandle is not IntPtr.Zero. `_closingGCHandle != IntPtr.Zero` above does that already, so it should not be needed here.